src/Monitoring/Sdk/GrafanaSerialization.cs

```diff
@@ -108,6 +118,118 @@ public static JObject SanitizeNotificationChannel(JObject notificationChannel)
             notificationChannel.Remove("updated");
             return notificationChannel;
         }
+
+        public static JObject ParameterizeDashboard(JObject dashboard, ICollection<Parameter> parameters, IEnumerable<string> environments, string activeEnvironment)
+        {
+            JObject parameterizedDashboard = new JObject(dashboard);
+
+            // These paths may be parameterized
+            string[] paths = new[]
+            {
+                "$.dashboard.panels[*].targets[*].azureMonitor.resourceGroup",
+                "$.dashboard.panels[*].targets[*].azureMonitor.resourceName",
+                "$.dashboard.panels[*].targets[*].azureLogAnalytics.resource",
+                "$.dashboard.panels[*].targets[*].azureLogAnalytics.workspace"
+            };
+
+            foreach (string path in paths)
+            {
+                IEnumerable<JToken> tokens = parameterizedDashboard.SelectTokens(path);
+
+                foreach (JToken token in tokens)
+                {
+                    string value = token.Value<string>();
+
+                    // Value Can technically be other JSON types (but the paths specified above
+                    // will always be strings). Because the object must be a "string" to store the
+                    // parameter info, supporting anything other than a string will require also
+                    // keeping type info. So only support strings for now.
+
+                    if (value == null)
+                    {
+                        throw new Exception("Unexpected token type. Value must be a string.");
+                    }
+
+                    Parameter p = parameters
+                        .Where(p => p.Values.ContainsKey(activeEnvironment) && p.Values[activeEnvironment] == value)
+                        .FirstOrDefault();
+
+                    if (p == null)
+                    {
+                        string name = Guid.NewGuid().ToString();
```

Comment:
That's... certainly a name.  It feels like we'd really want these to have names... unless we don't? How is someone supposed to know what to edit in the placeholder when it's just going to be some GUID goes to some other GUID?

Maybe the build should fail, put a placeholder in the parameters file, and force someone to choose a real name for it, rather than just create a big pile of meaningless GUID's that'll be a real pain to maintain later.

Reply:
Perhaps ugly, but as GUIDs they are unique values so brute force find-replace works well. 

What I eventually figured out: Export the dashboard, let it auto-generate any missing parameters. Fixup the names in the parameters file, then export the dashboard again. It will match-up the friendly names. 

I guess I could add a "rematch / rename" function that would look at an existing export file and rename parameters.

Reply:
That's why I was wodnering if it should just output the parameters file then error. Then you'd rename the new thing in the export, and export again.